### PR TITLE
Sip support

### PR DESCRIPTION
**Pull request by chaosloth**
### Summary

Add a mute and hangup buttons for non-WebRTC calls (i.e. Calls via SIP or PSTN). The default buttons expect the Voice SDK to be used as it calls the local method to mute and terminate calls, this plugin introduces a replacement UI buttons for the call canvas with a companion serverless functions to mute the Flex worker by modifying the conference participants. 

### Checklist

- [X] Tested changes end to end
- [X] Updated documentation
- [X] Requested one or more reviewers

### CODE REVIEW

**Comment by chaosloth** [comment]
@dremin hey mate, this is ready for review, incorporated changes from #342 and added support for HangUp

File: docs/docs/feature-library/sip-support.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+![SIP Support demo](/img/features/sip-support/demo.gif)
+
+# setup and dependencies
+
+## flex-config
+
+Within your `ui_attributes` file, the `sip-support` feature has 3 settings you may modify:
+
+- `true` - whether any functionality from this feature is enabled
+- `false` - default flex behaviour
```
Comment (dremin) [inline docs/docs/feature-library/sip-support.md:25]: As far as I could tell there was only the enabled setting, not 3 settings?

File: plugin-flex-ts-template-v2/src/feature-library/sip-support/custom-components/CustomHangupButton.tsx
Comment (dremin) [inline plugin-flex-ts-template-v2/src/feature-library/sip-support/custom-components/CustomHangupButton.tsx:1]: While this works for one hangup button, it does not address the hangup button in the TaskCanvasHeader. It also does not invoke [before/after]HangupCall action listeners. Could this instead be implemented as replacing the HangupCall action? In my testing the same code works when invoked that way.

**Review by dremin — CHANGES_REQUESTED** [review]
Thanks! I think we will want to implement the hangup support a bit differently to account for additional use cases, see the below comment.

File: docs/docs/feature-library/sip-support.md (hunk 1 above)
Comment (chaosloth) [reply docs/docs/feature-library/sip-support.md]: Updated

File: plugin-flex-ts-template-v2/src/feature-library/sip-support/custom-components/CustomHangupButton.tsx
Comment (chaosloth) [reply plugin-flex-ts-template-v2/src/feature-library/sip-support/custom-components/CustomHangupButton.tsx:1]: Gotcha -- I've updated it to an Action now replacing the HangupCall action

**Review by dremin — CHANGES_REQUESTED** [review]
This is looking pretty good, thanks for making those changes! Just a couple small things left to button this up, then we will be good to merge ☺️

**Comment by dremin** [comment]
Ignore the failed GitHub workflow-- it can't post its comment when running in a branch, but looking at the logs the lint checks found no issues, so we are all good 😄

File: docs/static/img/features/sip-support/demo.gif
[the remark this answers is not in the document]
Comment (chaosloth) [reply docs/static/img/features/sip-support/demo.gif:1]: Cool, now compressed to 1.9MB

File: plugin-flex-ts-template-v2/src/feature-library/sip-support/flex-hooks/actions/HangupCall.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+import { FlexAction } from '../../../../types/feature-loader';
+import {
+  getConferenceSidFromTask,
+  getLocalParticipantForTask,
+  isWorkerUsingWebRTC,
+} from '../../helpers/CallControlHelper';
+import CallControlService from '../../helpers/CallControlService';
+
+export const actionEvent = '';
```
[the remark this answers is not in the document]
Comment (chaosloth) [reply plugin-flex-ts-template-v2/src/feature-library/sip-support/flex-hooks/actions/HangupCall.ts]: Done :)

**Review by dremin — APPROVED** [review]
LGTM! 🎉

**Comment by lhamoudi** [comment]
Awesome!

**Pull request merged**